plugins/trivy/cmd/trivy/main.go

```diff
@@ -30,95 +30,20 @@ type newestVersions struct {
 }
 
 func main() {
-
-	concurrencyStr := os.Getenv("MAX_CONCURRENT_SCANS")
-	if concurrencyStr != "" {
-		var err error
-		maxConcurrentScans, err = strconv.Atoi(concurrencyStr)
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	numberToScanStr := os.Getenv("MAX_SCANS")
-	if numberToScanStr != "" {
-		var err error
-		numberToScan, err = strconv.Atoi(numberToScanStr)
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	ignoreUnfixedStr := os.Getenv("IGNORE_UNFIXED")
-	if ignoreUnfixedStr != "" {
-		ignoreUnfixedBool, err := strconv.ParseBool(ignoreUnfixedStr)
-		if err != nil {
-			panic(err)
-		}
-		if ignoreUnfixedBool {
-			extraFlags += "--ignore-unfixed"
-		}
-	}
-
-	err := util.RunCommand(exec.Command("trivy", "image", "--download-db-only"), "downloading trivy database")
-	if err != nil {
-		panic(err)
-	}
-	util.CheckEnvironmentVariables()
+	setEnv()
 	lastReport := image.GetLastReport()
 	ctx := context.Background()
 	images, err := image.GetImages(ctx)
 	if err != nil {
 		panic(err)
 	}
-
-	imagesToScan := make([]models.Image, 0)
-	for _, image := range images {
-		found := false
-
-		for _, report := range lastReport.Images {
-			if report.Name == image.Name && report.ID == image.ID {
-				found = true
-				break
-			}
-		}
-		if !found {
-			imagesToScan = append(imagesToScan, image)
-		}
-	}
-	imagesToKeep := make([]models.ImageDetailsWithRefs, 0)
-	sort.Slice(lastReport.Images, func(a, b int) bool {
-		return lastReport.Images[a].LastScan == nil || lastReport.Images[b].LastScan != nil && lastReport.Images[a].LastScan.Before(*lastReport.Images[b].LastScan)
-	})
-	for _, report := range lastReport.Images {
-		keep := false
-		for _, image := range images {
-			if report.Name == image.Name && report.ID == image.ID {
-				if len(imagesToScan) < numberToScan {
-					imagesToScan = append(imagesToScan, image)
-					break
-				}
-				keep = true
-				break
-			}
-		}
-		if keep {
-			imagesToKeep = append(imagesToKeep, report)
-		}
-	}
-	lastReport.Images = imagesToKeep
+	imagesToScan := getImagesToScan(images, lastReport.Images)
+	lastReport.Images = getImagesToKeep(images, lastReport, imagesToScan)
```

Comment:
can we log both `images` and `imagesToScan`?

Reply:
done